File: lib-segment/src/main/java/websiteschema/mpsegment/dict/IWord.java
package websiteschema.mpsegment.dict;

import websiteschema.mpsegment.concept.Concept;

//TODO: add concept info here
public interface IWord {

    public int getLog2Freq();

    public int getDomainType();

    public long getOccuredCount(String s);

    public long getOccuredSum();

    public POSArray getPOSArray();

    public int[][] getWordPOSTable();

    public int getWordLength();

    public String getWordName();

    public void incOccuredCount(String s);

    public void setDomainType(int i);

    public void setOccuredCount(String s, int i);

    public void setOccuredSum(int i);

    public int getWordMaxPOS();

    public String toDBFString();

    public Concept[] getConcepts();
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/dict/IWord.scala
package websiteschema.mpsegment.dict

import websiteschema.mpsegment.concept.Concept

trait IWord {

    def getLog2Freq() : Int

    def getDomainType() : Int

    def getOccuredCount(s: String) : Int

    def getOccuredSum() : Int

    def getPOSArray() : POSArray

    def getWordPOSTable() : Array[Array[Int]]

    def getWordLength() : Int

    def getWordName() : String

    def incOccuredCount(s: String)

    def setDomainType(i: Int)

    def setOccuredCount(s: String, i: Int)

    def setOccuredSum(i: Int)

    def getWordMaxPOS() : Int

    def toDBFString() : String

    def getConcepts() : Array[Concept]
}